### PR TITLE
[Search] documentation update

### PR DESCRIPTION
**Pull request by emmaindal**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
😅

File: docs/features/search/architecture.md [hunk 1]
```diff
@@ -6,9 +6,8 @@ description: Documentation on Search Architecture
 
 # Search Architecture
 
-> _This is a proposed architecture which has not been implemented yet. We are
-> still looking for feedback to improve the architecture to fit your use-case,
-> see [this open issue](https://github.com/backstage/backstage/issues/4078)._
+> _This is a proposed architecture which has not been implemented yet. Find our
+> milestones to follow our progress [here](./README.md)._
```
Comment (adamdmharvey) [inline docs/features/search/architecture.md:10]:
```suggestion
> milestones to follow our progress on the [Search Roadmap](./README.md#project-roadmap)._
```

A pet peeve of mine, but would it be possible to reword this?  (the dreaded avoiding "click here" 😁 
 https://www.nngroup.com/articles/top-ten-web-design-mistakes-of-2005/ )

I think this hash link will work...

**Review by realandersn — APPROVED** [review]
Nothing is ever easy... 😭

**Comment by emmaindal** [comment]
> Nothing is ever easy... 😭

+1 whats easy for you is never easy for me :PPPP

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4459-ppidpr8artzihmofvdhx6mzdbj6goyz0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6022a35798552d195c5a40a7

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4459-ppidpr8artzihmofvdhx6mzdbj6goyz0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6022a35798552d195c5a40a7

**Comment by emmaindal** [comment]
whut happened here 👀

**Comment by emmaindal** [comment]
hahah nvm, a  👻 were here

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4459-ppidpr8artzihmofvdhx6mzdbj6goyz0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6022a35798552d195c5a40a7

**Comment by OrkoHunter** [comment]
> hahah nvm, a 👻 was here

I saw it, I'm telling you - there were some unexpected changes, and now they are gone. I don't know, I saw it.

**Pull request merged**